modules/meta/arrow-meta-prototype/testing-plugin/src/main/kotlin/arrow/meta/plugin/testing/Compilation.kt

```diff
@@ -0,0 +1,65 @@
+package arrow.meta.plugin.testing
+
+import com.tschuchort.compiletesting.KotlinCompilation
+import com.tschuchort.compiletesting.SourceFile
+import org.assertj.core.api.Assertions.assertThat
+import java.io.File
+import java.nio.file.Paths
+import io.github.classgraph.ClassGraph
+
+enum class CompilationResult {
+  OK,
+  INTERNAL_ERROR,
+  COMPILATION_ERROR,
+  SCRIPT_EXECUTION_ERROR
+}
+
+data class CompilationData(
+  val sourceFileName: String,
+  val sourceContent: String,
+  val generatedFileContent: String,
+  val generatedClasses: ArrayList<String>,
+  val compilationResult: CompilationResult
+)
+
+fun testCompilation(compilationData: CompilationData) {
```

Comment:
Even for functions that effectful and return `Unit` I consider good practice for all public API's to always specify what they return so there is no ambiguity in the reader as to what type of inference came up with.

Reply:
I love that guideline!